flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/TaskExecutorPartitionTrackerImpl.java

```diff
@@ -141,12 +146,20 @@ void addPartition(ResultPartitionID resultPartitionId) {
             partitionIds.add(resultPartitionId);
         }
 
+        void addShuffleDescriptor(ShuffleDescriptor shuffleDescriptor) {
```

Comment:
nit: can we merge addPartition and addShuffleDescriptor into one method? For example:
        void addPartition(ShuffleDescriptor shuffleDescriptor) {
            partitionIds.add(shuffleDescriptor.getResultPartitionID());
            shuffleDescriptors.add(shuffleDescriptor);
        }